EIPS/eip-2935.md

```diff
@@ -24,38 +24,31 @@ A side benefit of this approach could be that it allows building/validating proo
 
 ## Specification
 
-| Parameter | Value |
-| - | - |
-| `FORK_TIMESTAMP` | TBD |
-| `HISTORY_STORAGE_ADDRESS` | `0x25a219378dad9b3503c8268c9ca836a52427a4fb`|
-| `HISTORY_SERVE_WINDOW` | `8192` |
-| `BLOCKHASH_OLD_WINDOW` | `256` |
+| Parameter                 | Value                                        |
+| ------------------------- | -------------------------------------------- |
+| `FORK_TIMESTAMP`          | TBD                                          |
+| `VERKLE_FORK_TIMESTAMP`   | TBD                                          |
+| `HISTORY_STORAGE_ADDRESS` | `0x25a219378dad9b3503c8268c9ca836a52427a4fb` |
+| `HISTORY_SERVE_WINDOW`    | `8192`                                       |
+| `BLOCKHASH_OLD_WINDOW`    | `256`                                        |
 
 This EIP specifies for storing last `HISTORY_SERVE_WINDOW` block hashes in a ring buffer storage of `HISTORY_SERVE_WINDOW` length.
 
-
-At the start of processing any block where `block.timestamp >= FORK_TIMESTAMP` (ie. before processing any transactions), update the history in the following way:
+At the start of processing any block where `block.timestamp >= FORK_TIMESTAMP` (ie. before processing any transactions), begin storing the blockhashes in the following way:
 
 ```python
 def process_block_hash_history(block: Block, state: State):
     if block.timestamp >= FORK_TIMESTAMP:
         state.insert_slot(HISTORY_STORAGE_ADDRESS, (block.number-1) % HISTORY_SERVE_WINDOW , block.parent.hash)
+```
 
-    # If this is the fork block, add the parent's direct `HISTORY_SERVE_WINDOW - 1` ancestors as well
-    if block.parent.timestamp < FORK_TIMESTAMP:
-        ancestor = block.parent
-        for i in range(HISTORY_SERVE_WINDOW - 1):
-            # stop at genesis block
-            if ancestor.number == 0:
-                break
+Note that from the `fork_block` (first block where `block.timestamp >= FORK_TIMESTAMP`), the implementation of `BLOCKHASH` does not change. The blockhashes of the last `BLOCKHASH_OLD_WINDOW` blocks can be retrieved as they previously were. `BLOCKHASH` will return `0` for blocks preceding the `BLOCKHASH_OLD_WINDOW`.
 
-            ancestor = ancestor.parent
-            state.insert_slot(HISTORY_STORAGE_ADDRESS, ancestor.number % HISTORY_SERVE_WINDOW, ancestor.hash)
-```
+Blockhashes from the `HISTORY_SERVE_WINDOW` can be retrieved directly from storage for blocks whose `block.number + 1 >= fork_block`.
```

Comment:
Can this be clarified? I think the idea is that if you want a block within `HISTORY_SERVE_WINDOW` but larger than `BLOCKHASH_OLD_WINDOW` you now call into the contract - right? Can this sentence then be updated such that it is clear that the contract should now be called and thus that `BLOCKHASH` will still put 0 on stack (even though the data is in the contract storage)